apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/VolatileResourceManager.cs

```diff
@@ -115,8 +115,15 @@ public void SinglePhaseCommit(SinglePhaseEnlistment singlePhaseEnlistment)
         {
             try
             {
-                ExecuteHelper.WithErrorTranslationAndProfiling(() =>
-                    _transaction?.Commit(), "VolatileResourceManager.Commit");
+                // If we have no changes, then just tell the outer transaction that everything is good.
+                // This can happen with a readonly transaction or a write transaction where we never
+                // executed any mutations.
```

Comment:
So we're certain that the Spanner team is happy never to hear about a read-only transaction? I can imagine it being "safe but a bit rude" (if that wastes server-side resources) - but I may very well be wrong.